data-science-onramp/data-ingestion/setup_test.py

```diff
@@ -0,0 +1,181 @@
+import os
+import re
+
+import uuid
+
+from google.cloud import dataproc_v1 as dataproc
+from google.cloud import storage
+
+import pytest
+
+
+# Set global variables
+PROJECT = os.environ['GCLOUD_PROJECT']
+REGION = "us-central1"
+ZONE = "us-central1-a"
+CLUSTER_NAME = f'setup-test-{uuid.uuid4()}'
+BUCKET_NAME = f'setup-test-code-{uuid.uuid4()}'
+
+BUCKET = None
+
+
+@pytest.fixture(autouse=True)
```

Comment:
From what I can tell, this fixture works without yielding anything, but you could also get rid of the autouse and instead yield the cluster name, which you'd then call as an argument in the definition of the test function (See below comment for the GCS example)

Afaik, it's the same result, different method. Not experienced enough with pytest to know what best practice in this case is.

Reply:
It looks like we need to yield to separate the setup code from the teardown code for the cluster. And, we do not need to yield the cluster name because it is a global variable that uses a uuid.